### PR TITLE
ESLint: Variables

### PR DESCRIPTION
**Pull request by gbbr**
ESLint [variables section](http://eslint.org/docs/rules/#variables).

### CODE REVIEW

File: scripts/apps/archive/controllers/UploadController.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ export function UploadController($scope, $q, upload, api, archiveService, sessio
                 item.upload.abort();
             }
         }
-        if (index !== undefined) {
+        if (index) {
```
Comment (petrjasek) [inline scripts/apps/archive/controllers/UploadController.js]: imo `index` can be `0`

**Comment by gbbr** [comment]
@petrjasek the problem is that a key inside an object defined as `undefined`, is still a defined key. As in, it exists in the prototype. I'm ok with removing this rule if you are ok with taking this risk. What do you think?

**Comment by gbbr** [comment]
You can read more [here](http://eslint.org/docs/rules/no-undefined).

**Comment by petrjasek** [comment]
I see. I can imagine keeping this but then I wouldn't disable `== null`, because those 2 together make things quite complex. but lets wait what others think.

**Comment by gbbr** [comment]
`=== null` is staying if that's what you mean. As for `== null` could be an issue because `undefined == null` :-)

**Comment by petrjasek** [comment]
I mean `== null` to check for both `null` and `undefined`. this would become:
`x === null || typeof x === 'undefined'`

**Comment by gbbr** [comment]
Yeah, but we have `_.isNil(v)` which does just that.

**Comment by gbbr** [comment]
@petrjasek I'm closing this. I don't feel very strongly about it and it's already used too much in our codebase. It's hard to factor it out and make all the tests pass and if you don't fully agree with it then I am ok with dropping it.

**Pull request closed without merging**